### PR TITLE
Add and configure ESLint and update configuration for Prettier

### PR DESCRIPTION
**Pull request by IvanZosimov**
**Description:**
In the scope of this PR, we are adding the ESLint tool and updating Prettier:
- ESLint dependencies were added;
- Configuration files `.eslintrc.js` and `.eslintignore` were added;
- Prettier dependency was updated;
- Configuration files `.prettiertrc.js` and `.prettierignore` were added, `.prettierrc.json` was removed;
- Commands in the `package.json` file were updated;
- Repository was formatted and linted according to the added configuration files;
- Unit-tests were updated;
- File `.gitattributes` was updated to force git to replace CRLF eof to LF eof;

**Check list:**
- [ ] Mark if documentation changes are required.
- [ ] Mark if tests were added or updated to cover the changes.

### CODE REVIEW

File: __tests__/version-utils.test.ts [hunk 1]
```diff
@@ -1,69 +1,93 @@
...
+      "The '1.0.0invalid' doesn't satisfy semantic versioning specification"
+    );
+  });
 
-    it("throw when a tag contains an valid unstable semantic version", () => {
-        expect(() => versionUtils.validateSemverVersionFromTag("v1.0.0-beta.1")).toThrowError(
-            "It is not allowed to specify pre-release version to update the major tag"
-        );
-    });
+  it('throw when a tag contains an valid unstable semantic version', () => {
```
Comment (MaksimZhukov) [inline __tests__/version-utils.test.ts:60]:
```suggestion
  it('throw when a tag contains a valid unstable semantic version', () => {
```

File: __tests__/version-utils.test.ts [hunk 2]
```diff
@@ -1,69 +1,93 @@
...
+  it("validate a tag containing an valid semantic version with 'v' prefix", () => {
+    expect(() =>
+      versionUtils.validateSemverVersionFromTag('v1.0.0')
+    ).not.toThrow();
+  });
 
-    it("validate a tag containing an valid semantic version with build metadata", () => {
-        expect(() => versionUtils.validateSemverVersionFromTag("v1.0.0+20130313144700")).not.toThrow();
-    });
+  it('validate a tag containing an valid semantic version with build metadata', () => {
```
Comment (MaksimZhukov) [inline __tests__/version-utils.test.ts:46]:
```suggestion
  it('validate a tag containing a valid semantic version with build metadata', () => {
```

File: __tests__/version-utils.test.ts [hunk 3]
```diff
@@ -1,69 +1,93 @@
...
+  it('validate a tag containing an valid semantic version', () => {
+    expect(() =>
+      versionUtils.validateSemverVersionFromTag('1.0.0')
+    ).not.toThrow();
+  });
 
-    it("validate a tag containing an valid semantic version with 'v' prefix", () => {
-        expect(() => versionUtils.validateSemverVersionFromTag("v1.0.0")).not.toThrow();
-    });
+  it("validate a tag containing an valid semantic version with 'v' prefix", () => {
```
Comment (MaksimZhukov) [inline __tests__/version-utils.test.ts:40]:
```suggestion
  it("validate a tag containing a valid semantic version with 'v' prefix", () => {
```

File: __tests__/version-utils.test.ts [hunk 4]
```diff
@@ -1,69 +1,93 @@
...
+    ).toBeFalsy();
+  });
 });
 
-describe("validateSemverVersionFromTag", () => {
-    it("validate a tag containing an valid semantic version", () => {
-        expect(() => versionUtils.validateSemverVersionFromTag("1.0.0")).not.toThrow();
-    });
+describe('validateSemverVersionFromTag', () => {
+  it('validate a tag containing an valid semantic version', () => {
```
Comment (MaksimZhukov) [inline __tests__/version-utils.test.ts:34]:
```suggestion
  it('validate a tag containing a valid semantic version', () => {
```

**Review by MaksimZhukov — APPROVED** [review]

**Review by panticmilos — APPROVED** [review]

**Review by dmitry-shibanov — APPROVED** [review]

**Comment by CicDiplomacy** [comment]
name: Release new action version

on:
  release:
    types: [released]
  workflow_dispatch:
    inputs:
      TAG_NAME:
        description: 'Tag name that the major tag will point to'
        required: true

env:
  TAG_NAME: ${{ github.event.inputs.TAG_NAME || github.event.release.tag_name }}
defaults:
  run:
    shell: pwsh

jobs:
  update_tag:
    name: Update the major tag to include the ${{ github.event.inputs.TAG_NAME || github.event.release.tag_name }} changes
    # Remember to configure the releaseNewActionVersion environment with required approvers in the repository settings
    environment:
      name: releaseNewActionVersion
    runs-on: ubuntu-latest
    permissions:
      contents: write
    steps:
      - uses: actions/checkout@v3

      - name: Update the ${{ env.TAG_NAME }} tag
        id: update-major-tag
        uses: ./
        with:
          source-tag: ${{ env.TAG_NAME }}
          slack-webhook: ${{ secrets.SLACK_WEBHOOK }}

**Pull request merged**